src/treemodel/document.js

```diff
@@ -76,6 +85,16 @@ CKEDITOR.define( [
 		}
 
 		/**
+		 * Returns current selection done on this document.
+		 *
+		 * @readonly
+		 * @property {treeModel.Selection}
+		 */
+		get selection() {
+			return this._selection;
+		}
+
+		/**
```

Comment:
Why not simply `this.selection = new Selection();`? It is marked as `@readonly` anyway.